include/zephyr/rtio/rtio.h

```diff
@@ -1075,9 +1051,8 @@ static inline int rtio_sqe_rx_buf(const struct rtio_iodev_sqe *iodev_sqe, uint32
 
 			if (rc == 0) {
 				*buf_len = num_blks * blk_size;
-				map_entry->state = RTIO_MEMPOOL_ENTRY_STATE_ALLOCATED;
-				map_entry->block_idx = __rtio_compute_mempool_block_index(r, *buf);
-				map_entry->block_count = num_blks;
+				mutable_sqe->buf = *buf;
```

Comment:
I'm wondering what happened with the mempool situation here, seems like a complete change to what you had?

Reply:
Something isn't quite right, it is not working well with high data rates.